test/suites/backup/auto-recovery.robot

```diff
@@ -0,0 +1,68 @@
+*** Settings ***
+Documentation       Tests related to auto-recovery functionality
+
+Resource            ../../resources/microshift-host.resource
+Resource            ../../resources/microshift-process.resource
+Library             Collections
+Library             DateTime
+Library             ../../resources/libostree.py
+
+Suite Setup         Setup
+Suite Teardown      Teardown
+
+
+*** Test Cases ***
+Backups Are Created Successfully With Expected Schema
+    [Documentation]    Verify that backups intended for auto-recovery are created successfully
+    ...    and they are named with expected schema of DATETIME_DEPLOY-ID for ostree/bootc and DATETIME_MICROSHIFT-VERSION.
+    Stop MicroShift
+    Command Should Work    microshift backup --auto-recovery /var/lib/microshift-auto-recovery
+    ${backup_name}=    Command Should Work    ls /var/lib/microshift-auto-recovery
+    Verify Backup Name    ${backup_name}
+
+    [Teardown]    Run Keywords
```

Comment:
What's the main goal of putting this in Teardown? 
If we want to leave MicroShift running, we should probably remove `rm` command into the main function.

Reply:
I don't understand. It starts microshift because it was stopped. It removes the dir to clean up.

Reply:
We need to make sure MicroShift is running after this. Let's test it by putting `false` command in the first statement to verify that `Start MicroShift` is "always" executed.

Reply:
I verified following and even if the 1st command fails, the `Start MicroShift` will still run. I changed it though for explicitness 
```
    [Teardown]    Run Keywords
    ...    Command Should Work    false
    ...    AND
    ...    Start MicroShift
```